worker/caasoperator/commands/container-spec-set.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package commands
+
+import (
+	"io/ioutil"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/names.v2"
+	"gopkg.in/yaml.v2"
+)
+
+// ContainerspecSetCommand implements the container-spec-set command.
+type ContainerspecSetCommand struct {
+	cmd.CommandBase
+	ctx Context
+
+	specFile cmd.FileVar
+	unitName string
+}
+
+// NewContainerspecSetCommand makes a container-spec-set command.
+func NewContainerspecSetCommand(ctx Context) (cmd.Command, error) {
+	return &ContainerspecSetCommand{ctx: ctx}, nil
+}
+
+func (c *ContainerspecSetCommand) Info() *cmd.Info {
+	doc := `
+Sets configuration data to use for a container.
+By default, the spec applies to all units for the
+application. However, if a unit name is specified,
+the spec is used for just that unit.
+`
+	return &cmd.Info{
+		Name:    "container-spec-set",
+		Args:    "--file <spec yaml> [--unit <unit-name>]",
+		Purpose: "set container spec information",
+		Doc:     doc,
+	}
+}
+
+func (c *ContainerspecSetCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.specFile.SetStdin()
+	c.specFile.Path = "-"
+	f.Var(&c.specFile, "file", "file containing container spec")
+	f.StringVar(&c.unitName, "unit", "", "set this spec for the specified unit")
+}
+
+func (c *ContainerspecSetCommand) Init(args []string) error {
+	if c.unitName != "" && !names.IsValidUnit(c.unitName) {
+		return errors.NotValidf("unit name %q", c.unitName)
+	}
+	return cmd.CheckEmpty(args)
+}
+
+func (c *ContainerspecSetCommand) Run(ctx *cmd.Context) error {
+	specYaml, err := c.handleSpecFile(ctx)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return c.ctx.SetContainerSpec(specYaml, c.unitName)
+}
+
+func (c *ContainerspecSetCommand) handleSpecFile(ctx *cmd.Context) (string, error) {
+	file, err := c.specFile.Open(ctx)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	defer file.Close()
+
+	rawYaml, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if len(rawYaml) == 0 {
+		return "", errors.New("container spec file required")
+	}
+
+	var data map[string]interface{}
```

Comment:
Sorry I missed this before - why are we mandating YAML here? If we even need it at all, it would be better to put it in the SetContainerSpec method, as potentially other things might want to set the spec.

Reply:
We generally try and do such syntactic validation close to the source, but yeah, probably no need to mandate a format. Later we should add provider specific validation.